### DISCUSSION TITLE
Flows option to process multiple events one after another, not all at once

### DISCUSSION THREAD

**Discussion by luciandavidescu** [directus/directus#17230] [Feature Requests]
Events pertain to either one item or more, but with bulk operations the workflow becomes complicated, with some things quite impossible, as there's no apparent substitute to `foreach`. It would be much more straightforward if there was this option on the flows' "trigger", to send events not as a group but in succession.

**Comment by rijkvanzanten** [comment]
What a timing! @Nitwel just implemented this, which was merged 2 hours ago 🙂 

https://github.com/directus/directus/pull/16773

**Comment by luciandavidescu** [comment]
In the mean time I figured that, to achieve the purpose, I can iterate a flow over itself and just remove the first element on each run.

However I also waited to see how this works and... it's not quite obvious yet. So in a 'trigger flow' I tried parallel, serial and batch, but every time the exact same payload (either with multiple keys or just one) gets through.

So how exactly is this supposed to work?

**Reply by luciandavidescu** [reply]
Solved on Discord by @Attacler/Bart

For known array lengths, in _payload_, it can simply be wrapped like this:

```
[
    {
        "one": 1
    },
    {
        "two": 2
    },   
 {
        "three": 3
    }
]
```

For unknown array lengths, it only works with raw edit, to skip json validation, with something like `"{{items}}"`
